src/asl.ts

```diff
@@ -1090,7 +1185,74 @@ export class ASL {
         });
       }
     } else if (isBinaryExpr(expr)) {
-      // TODO
+      if (expr.op === "&&" || expr.op === "||") {
+        return {
+          default: {
+            Type: "Choice",
+            Choices: [{ ...ASL.toCondition(expr), Next: "assignTrue" }],
+            Default: "assignFalse",
+          },
+          assignTrue: {
+            Type: "Pass",
+            Result: true,
+            ...props,
+          },
+          assignFalse: {
+            Type: "Pass",
+            Result: false,
+            ...props,
+          },
+        };
+      } else if (expr.op === "??") {
+        // literal ?? anything
+        if (isLiteralExpr(expr.left)) {
+          if (
+            isNullLiteralExpr(expr.left) ||
+            isUndefinedLiteralExpr(expr.left)
+          ) {
+            return {
+              Type: "Pass",
+              ...ASL.toTaskInput(expr.right),
+              ...props,
+            };
+          } else {
+            return {
+              Type: "Pass",
+              ...ASL.toTaskInput(expr.left),
+              ...props,
+            };
+          }
+        }
+        const left = ASL.toJsonPath(expr.left);
+        return {
+          default: {
+            Type: "Choice",
+            Choices: [
+              {
+                ...ASL.and(ASL.isPresent(left), ASL.isNotNull(left)),
+                Next: "takeLeft",
+              },
+            ],
+            Default: "takeRight",
+          },
+          takeLeft: {
+            Type: "Pass",
+            InputPath: left,
+            ...props,
+          },
+          takeRight: {
+            Type: "Pass",
+            ...ASL.toTaskInput(expr.right),
+            ...props,
+          },
```

Comment:
If these get re-written, then I totally understand why you added that pass to re-write state names - isolation.